contracts/croncat-manager/src/contract.rs

```diff
@@ -528,8 +530,21 @@ pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractE
                     .iter()
                     .map(|(idx, failure)| Attribute::new(format!("action{}_failure", idx), failure))
                     .collect();
-                // TODO: agent: on_task_executed?
-                Ok(finalize_task(deps, queue_item)?.add_attributes(failures))
+                let config = CONFIG.load(deps.storage)?;
+                //todo: fix is_block_slot param after boundary fix
+                let complete_msg = create_task_completed_msg(
+                    &deps.querier,
+                    &config,
+                    queue_item.task.task_hash.clone(),
+                    &queue_item.agent_addr,
+                    matches!(
+                        queue_item.task.interval,
+                        Interval::Block(_) | Interval::Immediate
```

Comment:
```suggestion
                        Interval::Block(_) | Interval::Immediate | Interval::Once
```
Once is also always should be a cron

Reply:
We need block interval here @Buckram123

Reply:
nvm it could be just
 ```
!matches!(
                        queue_item.task.interval,
                        Interval::Cron(_)
                        ),
  ```